client/containers/DashboardSubmissions/VerdictDialog.tsx

```diff
@@ -0,0 +1,154 @@
+				</p>
+				<WorkflowTextEditor
+					initialContent={customEmailText}
+					onContent={setCustomEmailText}
+					placeholder="Specify message to pub author(s)."
+				/>
+			</div>
+			<div className={Classes.DIALOG_FOOTER}>
+				<div className={Classes.DIALOG_FOOTER_ACTIONS}>
+					<Checkbox
+						className="email-toggle"
+						checked={shouldSendEmail}
+						disabled={props.isHandlingSubmission}
+						onChange={(e) => {
+							setShouldSendEmail((e.target as HTMLInputElement).checked);
+						}}
+						label="Notify authors by email"
+					/>
+					<Button onClick={props.onClose} disabled={props.isHandlingSubmission}>
+						Cancel
+					</Button>
+					<Button
+						onClick={() => props.handleSubmission(customEmailText, shouldSendEmail)}
+						loading={props.isHandlingSubmission}
+						intent="primary"
+					>
+						Email & {props.actionTitle}
+					</Button>
+				</div>
+			</div>
+		</>
+	);
+};
+
+type PostSubmitBodyProps = {
+	completedName: string;
+	onClose: () => unknown;
+	isHandlingSubmission: boolean;
+};
+const PostSubmitBody = (props: PostSubmitBodyProps) => {
+	return (
+		<>
+			<div className={Classes.DIALOG_BODY}>
+				<Callout intent="success" title={`Submission ${props.completedName}!`}>
+					You successfully {props.completedName} the submission!
+				</Callout>
+			</div>
+			<div className={Classes.DIALOG_FOOTER}>
+				<div className={Classes.DIALOG_FOOTER_ACTIONS}>
+					<Button onClick={props.onClose} disabled={props.isHandlingSubmission}>
+						Close
+					</Button>
+				</div>
+			</div>
+		</>
+	);
+};
+
+const VerdictDialog = (props: Props) => {
+	const [isHandlingSubmission, setIsHandlingSubmission] = useState(false);
+	const [updatedSubmission, setUpdatedSubmission] = useState(null);
+	const [submissionError, setSubmissionError] = useState(null);
+	const handleSubmission = (customEmailText: DocJson, shouldSendEmail: boolean) => {
+		setIsHandlingSubmission(true);
+		apiFetch('/api/submissions', {
+			method: props.apiMethod,
+			body: JSON.stringify({
+				id: props.pub.submission.id,
+				skipEmail: !shouldSendEmail,
+				...(customEmailText && { customEmailText }),
+				...(props.status && { status: props.status }),
+			}),
+		})
+			.then((submissionRes) => {
+				setUpdatedSubmission(submissionRes);
+				setIsHandlingSubmission(false);
+			})
+			.then(() => props.onJudgePub(props.pub.id, props.status))
+			.catch((err) => {
+				setSubmissionError(err);
+				setIsHandlingSubmission(false);
+			});
+	};
+	return (
+		<Dialog
+			lazy={true}
+			title={props.actionTitle}
+			className="verdict-dialog"
+			isOpen={props.isOpen}
+			onClose={props.onClose}
+		>
+			{submissionError ? (
+				<Callout intent="warning" title="There was an error creating this submission." />
+			) : updatedSubmission ? (
+				<PostSubmitBody
+					isHandlingSubmission={isHandlingSubmission}
+					completedName={props.completedName}
+					onClose={props.onClose}
+				/>
+			) : (
+				<PreSubmissionBody
```

Comment:
Nit: if this is called `PreSubmissionBody`, I think the other component should be called `PostSubmissionBody`. I also wonder if a better noun than "submission" could be used here to refer to the act of submitting the form — it's uniquely ambiguous here because we're dealing with `Submission` objects as well :)